docs/ai_flags.md

```diff
@@ -0,0 +1,149 @@
+`.aiFlags = AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_TRY_TO_FAINT | AI_FLAG_CHECK_VIABILITY`
+
+# What AI Flags does pokeemerald-expansion have?
+This section lists all of expansion’s AI Flags and briefly describes the effect they have on the AI’s behaviour. In all cases, please check the corresponding function or surrounding code around their implementation for more details. Some of these functions are vanilla, some share a name with vanilla but have been modified to varying degrees, and some are completely new.
+
+## `AI_FLAG_CHECK_BAD_MOVE` 
+AI will avoid using moves that are considered poor in its current situation. The following list is not exhaustive. The AI will be disincentivized to:
+* Use powder moves into an immune target
+* Attack into invulnerability (Fly etc.) poorly
+* Use moves that are not very effective or worse
+* Use moves that are blocked or made worse by opposing abilities (don’t Poison Magic Guard, priority into Queenly Majesty, etc.) or partner abilities (Lightningrod redirect etc.)
+* Use moves that are negated by terrain
+* Use move effects that are poor (Attack buffs without a physical move, Charge if already charged, stat buffs if stats maxed, Roar when the opponent has one mon left, etc.)
+* Use moves that are not good in current weather, or set weather when partner is
+* Copy negative stat changes
+* Sucker Punch into predicted status move
+* Choice Lock into a poor move
+
+## `AI_FLAG_TRY_TO_FAINT`
+AI will prioritize KOing the player if able rather than using status moves. Will prioritize using a move that can OHKO the player. If the player can KO the AI’s mon and the AI’s mon is slower, prioritize priority moves (this does not prevent the AI from switching out instead).
+
+## `AI_FLAG_CHECK_VIABILITY`
+AI will choose the best available attacking move among its options to hit the player. Moves are priorities that take fewer hits to KO, don’t take a turn to charge, have more accuracy, and have a better effect. Good moves effects are also scored depending on current context.
+
+## `AI_FLAG_SETUP_FIRST_TURN`
+AI will prioritize using setup moves on the first turn. These include stat buffs, field effects, status moves, etc.
+
+## `AI_FLAG_RISKY`
+AI will generally behave more recklessly. This AI enables the following behaviour:
+* Always assume the highest damage roll when scoring moves
+* Blindly Mirror Coat / Counter based on the player mon’s species higher attacking stat
+* Moves with Recoil if they miss are not treated differently even if accuracy is lowered
+* Prioritize maximizing damage from moves at the cost of accuracy
+* Prioritize moves with low change strong effects (Ancient Power etc., check `AI_Risky` function for full list)
+* Switch offensively mid battle rather than defensively (if using `AI_FLAG_SMART_MON_CHOICES`)
+* Prioritize Explosion moves
+
+## `AI_FLAG_PREFER_STRONGEST_MOVE`
+Adds score bonus to any move the AI has that either OHKOs or 2HKOs the player. 
```

Comment:
This is a weaker form of `AI_FLAG_TRY_TO_FAINT` which I would add as an additional comment.